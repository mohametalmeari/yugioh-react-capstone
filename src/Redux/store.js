import { configureStore } from '@reduxjs/toolkit';
import typesReducer from './types/typesSlice';
import detailsReducer from './details/detailsSlice';

const store = configureStore({
  reducer: {
    types: typesReducer,
    details: detailsReducer,
  },
});

export default store;
