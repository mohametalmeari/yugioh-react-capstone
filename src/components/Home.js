import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import { getTypes } from '../Redux/types/typesSlice';
import CardsType from './CardsType';
import headerImg from '../images/5.png';
import { RefreshIcon } from '../icons';

const Home = () => {
  const { typesList, isLoading } = useSelector((state) => state.types);
  const dispatch = useDispatch();
  useEffect(() => {
    if (typesList.length === 0) {
      dispatch(getTypes());
    }
  }, [typesList.length, dispatch]);

  return (
    <div className="container">
      <nav>
        <NavLink
          to="/"
          onClick={() => dispatch(getTypes())}
        >
          <RefreshIcon />
        </NavLink>
        <span>All Cards</span>
      </nav>
      <header>
        <img data-testid="img-1" className="header-img" src={headerImg} alt="Logo" />
        <div className="streched-gap" />
        <div className="header-content">
          <h1>
            YU-GI-OH!
          </h1>
          <p>
            {typesList.reduce((sum, obj) => sum + obj.count, 0)}
            {' '}
            cards
          </p>
        </div>
      </header>
      <main>
        <h2 className="list-title">STATS BY TYPE</h2>
        <section className="home-list">
          {isLoading && <p>Loading...</p>}
          {!isLoading && typesList.map((item) => (
            <CardsType
              key={item.typeId}
              typeId={item.typeId}
              typeName={item.type}
              typeCount={item.count}
            />
          ))}
        </section>
      </main>
    </div>
  );
};
export default Home;
