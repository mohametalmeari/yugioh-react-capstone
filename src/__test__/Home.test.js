import { render, screen, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import Home from '../components/Home';

describe('Home App', () => {
  afterEach(() => cleanup());
  test('renders titles', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </MemoryRouter>,
    );
    expect(screen.getByText(/All Cards/i)).toBeInTheDocument();
    expect(screen.getByText(/YU-GI-OH!/i)).toBeInTheDocument();
    expect(screen.getByText(/STATS BY TYPE/i)).toBeInTheDocument();
  });
  test('renders header image', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </MemoryRouter>,
    );
    expect(screen.getByTestId('img-1')).toBeInTheDocument();
  });
  test('renders Loading', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </MemoryRouter>,
    );
    expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
  });
  test('renders Snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
