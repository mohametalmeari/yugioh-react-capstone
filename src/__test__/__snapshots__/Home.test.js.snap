// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home App renders Snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <nav>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0l3.181 3.183a8.25 8.25 0 0013.803-3.7M4.031 9.865a8.25 8.25 0 0113.803-3.7l3.181 3.182m0-4.991v4.99"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </a>
      <span>
        All Cards
      </span>
    </nav>
    <header>
      <img
        alt="Logo"
        class="header-img"
        data-testid="img-1"
        src="5.png"
      />
      <div
        class="streched-gap"
      />
      <div
        class="header-content"
      >
        <h1>
          YU-GI-OH!
        </h1>
        <p>
          0
           
          cards
        </p>
      </div>
    </header>
    <main>
      <h2
        class="list-title"
      >
        STATS BY TYPE
      </h2>
      <section
        class="home-list"
      >
        <p>
          Loading...
        </p>
      </section>
    </main>
  </div>
</div>
`;
